ui/src/app/pages/homepage/data-access-requirements.tsx

```diff
@@ -293,10 +293,29 @@ const selfBypass = async(spinnerProps: WithSpinnerOverlayProps, reload: Function
   reload();
 };
 
-const Refresh = () => <Button
+const syncExternalModules = async() => {
+  const aborter = new AbortController();
+  try {
+    await Promise.all([
+      profileApi().syncTwoFactorAuthStatus({signal: aborter.signal}),
+      profileApi().syncComplianceTrainingStatus({signal: aborter.signal}),
+      profileApi().syncEraCommonsStatus({signal: aborter.signal}),
+    ]);
+  } catch (e) {
+    if (!isAbortError(e)) { throw e; }
+  } finally {
+    aborter.abort();
+  }
+};
+
+const Refresh = (props: {showSpinner: Function}) => <Button
     type='primary'
     style={styles.refreshButton}
-    onClick={() => location.reload()} >
+    onClick={() => {
+      props.showSpinner();
+      syncExternalModules();
```

Comment:
You probably need to make this callback async, and then add an await here. As written I think this is non-deterministic:

- kick off a bunch of requests (don't wait for them to finish)
- reload the page (requests may not have even been sent yet, unclear - at the very least they haven't finished)

Reply:
Thanks ... I was having trouble with useEffect and async, so I forgot it worked with onClick.  fixed.  (also for selfBypass)